### PR TITLE
docs(servers): update server conf files for offline-first per #645

### PR DESCRIPTION
**Pull request by gihrig**
Addresses [#645 comment](https://github.com/mxstbr/react-boilerplate/issues/645#issuecomment-263550516) re caching.

While reviewing an unrelated issue I realized a need for updated server configs to fully support offline-first operation. This PR implements those changes as I believe they should be.

**I have not tested these configurations**

It would be great if others could review, test and confirm these changes are correct.

Additional resources:

Apache
http://stackoverflow.com/a/33287352/522756

Nginx
https://www.nginx.com/blog/nginx-caching-guide/ (search pragma)

### CODE REVIEW

**Comment by Dattaya** [comment]
@ajarbol, could you test this nginx config to see whether it works as expected or not?

**Comment by gihrig** [comment]
@ajarbol (any others knowledgeable in Nginx) This PR is a foundation for #1393 on the same subject, but security best practice. That needs review also.

File: docs/general/server-configs.md [hunk 1]
```diff
@@ -2,17 +2,19 @@
 
 ## Apache
 
-This boilerplate includes a `.htaccess` file that does two things:
+This boilerplate includes an `app/.htaccess` file that does three things:
 
 1. Redirect all traffic to HTTPS because ServiceWorker only works for encrypted
    traffic.
 1. Rewrite all pages (e.g. `yourdomain.com/subpage`) to `yourdomain.com/index.html`
    to let `react-router` take care of presenting the correct page.
```
Comment (mxstbr) [inline docs/general/server-configs.md]: This empty line needs to be removed, it is significant in markdown.

Comment (gihrig) [inline docs/general/server-configs.md]: I think you're referring to line 11? It's removed.

**Comment by gihrig** [comment]
Apparently, I am able to approve my own PR. But this seems just a bit 'over the top'.

Still I think this change is better included in 3.4 than left out, even if untested. These server configs are recommendations only, too small really to just copy-paste and go. The boilerplate does have some specific web server config requirements. Isn't it better to give folks our best effort, even if not quite right, than nothing?

So @mxstbr or @Dattaya, I'm suggesting that lacking a tested approval, we merge as is for 3.4. Agree?

**Pull request merged**